src/orchestrator-process/src/lib.rs

```diff
@@ -161,7 +161,14 @@ impl NamespacedOrchestrator for NamespacedProcessOrchestrator {
                             process.kill();
                         } else {
                             // Existing non-dead process, so we don't create a new one
-                            processes.push(port_metadata);
+                            for port in port_metadata.values() {
+                                if !self.port_allocator.mark_allocated(*port) {
+                                    // Somehow we've re-allocated an already used port which
+                                    // shouldn't be possible. So we just kill the process and panic.
```

Comment:
It seems like it _might_ be possible if the orchestrator is not presented with all of the previously-created services on startup in the same order as they were originally created? But that doesn't happen presently, I think, and seems fine to ignore for now!